jerry-libc/include/assert.h

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2016 Samsung Electronics Co., Ltd.
+ * Copyright 2016 University of Szeged.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef JERRY_LIBC_ASSERT_H
+#define JERRY_LIBC_ASSERT_H
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif /* !__cplusplus */
+
+#ifndef NDEBUG
+#define assert(x) \
```

Comment:
Because `assert.h` and the `assert` macro (and its depencendcy on the `NDEBUG` macro) are part of the C standard, while `LIBC_ASSERT` and `LIBC_NDEBUG` are not. It is not cost declaring things the standard-conforming way but has the benefit that we can expose it in a public header and make it useful to libc users.

I have to admit that I have future plans with it. I came to realize the embedders might need a proper `assert`.